smartsim/settings/base.py

```diff
@@ -448,12 +455,16 @@ def add_exe_args(self, args: t.Union[str, t.List[str]]) -> None:
         :type args: str | list[str]
         :raises TypeError: if exe args are not strings
         """
+
+        if not isinstance(args, (list, str)):
+            raise TypeError("All elements in the list should be of type str")
+
```

Comment:
Should we be using the _build_exe_args() function in this method instead of repeating the logic in that method?